engine/classes/Elgg/Database/SubtypeTable.php

```diff
@@ -317,12 +255,46 @@ function update($type, $subtype, $class = '') {
 			':class' => $class,
 			':id' => $id,
 		];
-		$success = $this->db->updateData($sql, false, $params);
-	
-		if ($success && isset($this->cache[$id])) {
-			$this->cache[$id]->class = $class;
+		if (!$this->db->updateData($sql, false, $params)) {
+			return false;
 		}
-	
-		return $success;
+
+		$this->invalidateCache();
+
+		return true;
+	}
+
+	/**
+	 * Empty the cache. Also invalidates the boot cache and memcache
+	 *
+	 * @return void
+	 */
+	private function invalidateCache() {
+		$this->cache = null;
+		_elgg_services()->boot->invalidateCache();
+		if (is_memcache_available()) {
+			(new \ElggMemcache('new_entity_cache'))->clear();
+		}
```

Comment:
@hypeJunction we just clear the memcache. That way we don't have to jump through the subtype code before returning a memcache entity.